core/types/accounts/account.go

```diff
@@ -38,35 +41,68 @@ func NewAccount() Account {
 	return a
 }
 
-func (a *Account) encodingLength(forStorage bool) uint {
-	var structLength uint
+func (a *Account) EncodingLengthForStorage() uint {
+	var structLength uint = 1 // 1 byte for fieldset
 	var nonContract = a.IsEmptyCodeHash() && a.IsEmptyRoot()
```

Comment:
Please do not use `nonContract` variable and instead treat all the fields separately, like in the decoding